### PR TITLE
Add NFS/NTS interface to desisurvey scheduler

### PR DESCRIPTION
**Pull request by schlafly**
Adds Next Tile Selector functionality to the desisurvey scheduler.

The key addition of this routine to desisurvey is the NTS.py module that connects to the DOS NFS.py module to give the online system access to the desisurvey scheduler.

A few minor changes were made outside NTS.py:
- Planner and Scheduler save(...) methods were made atomic
- scheduler.next_tile(...) now accepts an program argument.  Tiles are selected from this program.
- next_tile also takes a skylevel argument, which is presently ignored.
- tiles.index(...) now accepts a default False return_mask argument.  If True, instead of raising an exception if an invalid tile ID is sent, the function returns a mask, indicating which IDs were valid.

### CODE REVIEW

File: py/desisurvey/NTS.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        scheduler = desisurvey.scheduler.Scheduler(
+            restore='scheduler_end_{}.fits')
+    else:
+        planner = desisurvey.plan.Planner(rules)
+        scheduler = desisurvey.scheduler.Scheduler()
+    # restore: maybe check directory, and restore if file present?  EFS
+    # planner.save(), scheduler.save()
+    # planner.restore(), scheduler.restore()
+    import dateutil.parser
+    planner.afternoon_plan(dateutil.parser.parse(night).date(),
```
Comment (dkirkby) [inline py/desisurvey/NTS.py]: I suggest using `desisurvey.utils.get_date(night)` here so we only have one entry point for night -> date conversion.

File: py/desisurvey/NTS.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+import desisurvey.rules
+import desisurvey.plan
+import desisurvey.scheduler
+import desisurvey.etc
+import datetime
+
+
+class NTS():
+    def __init__(self, obsplan, fiber_assign_dir, defaults={}, night=None):
+        """The caller has checked that the obsplan file exists and that
```
Comment (dkirkby) [inline py/desisurvey/NTS.py]: Please expand to a full docstring for readthedocs

File: py/desisurvey/NTS.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        self.scheduler.update_tiles(self.planner.tile_available,
+                                    self.planner.tile_priority)
+        self.scheduler.init_night(self.night, use_twilight=True)
+        self.ETC = desisurvey.etc.ExposureTimeCalculator()
+
+    def next_tile(self, mjd=None, skylevel=None, transparency=None,
+                  seeing=None, program=None, lastexp=None, fiber_assign=None,
+                  previoustiles=None):
+        """
+        select the next tile
```
Comment (dkirkby) [inline py/desisurvey/NTS.py]: Please expand to a full docstring for readthedocs

File: py/desisurvey/NTS.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+                     'esttime': exptime*days_to_seconds,
+                     'maxtime': maxtime*days_to_seconds,
+                     'fiber_assign': fiber_assign,
+                     'foundtile': True}
+
+        return selection
+
+
+def afternoon_plan(night=None, lastnight=None):
+    if night is None:
```
Comment (dkirkby) [inline py/desisurvey/NTS.py]: Please add a docstring for readthedocs

File: py/desisurvey/NTS.py (hunk 1 above)
Comment (schlafly) [reply py/desisurvey/NTS.py]: Should be resolved in latest commits.

**Comment by schlafly** [comment]
These changes should resolve the ra/dec <-> dec/ra mixup and improved the NTS documentation.

**Comment by dkirkby** [comment]
The changes look good, thanks!  I am restarting the travis job that failed since it looks like a network glitch.

**Comment by dkirkby** [comment]
Tests passing now.

**Pull request merged**